### PR TITLE
[Docs] Add note about AIR to Tune resources guide

### LINKED ISSUE #31067: [Tune/docs] Mention AIR concepts in Tune resources guide, throw an exception in `tune.with_resources`
### Description

We should mention that `tune.with_resources` is not supposed to be used with AIR `Trainers` in the Tune resources guide.

Example of confusion - https://discuss.ray.io/t/using-fractional-gpu-with-torchtrainer-and-tuner-api/8598

### Link

https://docs.ray.io/en/latest/tune/tutorials/tune-resources.html

### PR DESCRIPTION
**Pull request by Yard1**
Signed-off-by: Antoni Baum <antoni.baum@protonmail.com>

## Why are these changes needed?

Adds a warning to Tune Resources Guide to let users know that AIR Trainers are configured through `ScalingConfig` and not `tune.with_resources`. See attached issue for more context.

## Related issue number

Closes https://github.com/ray-project/ray/issues/31067

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/tune/tutorials/tune-resources.rst [hunk 1]
```diff
@@ -18,6 +18,10 @@ of CPUs (cores) on your machine.
     )
     results = tuner.fit()
 
+.. warning::
```
Comment (justinvyu) [inline doc/source/tune/tutorials/tune-resources.rst:21]: Can we move this warning below the next paragraph? We should introduce `tune.with_resources` before having this warning.

File: doc/source/tune/tutorials/tune-resources.rst [hunk 2]
```diff
@@ -18,6 +18,10 @@ of CPUs (cores) on your machine.
...
+
+    ``tune.with_resources`` cannot be used with :ref:`Ray AIR Trainers <air-trainers>`. If you are passing a Trainer to a Tuner, specify the resource requirements in the Trainer instance using :class:`~ray.air.config.ScalingConfig`. The general principles outlined below still apply.
+
 You can override this per trial resources with ``tune.with_resources``. Here you can
 specify your resource requests using either a dictionary or a
```
Comment (justinvyu) [inline doc/source/tune/tutorials/tune-resources.rst:26]:
```suggestion
specify your resource requests using either a dictionary, a :class:`~ray.air.config.ScalingConfig`, or a
```

File: doc/source/tune/tutorials/tune-resources.rst [hunk 3]
```diff
@@ -60,10 +64,9 @@ object. In any case, Ray Tune will try to start a placement group for each trial
     )
     results = tuner.fit()
 
+Similarly, you can also use the ``ScalingConfig`` to specify trial resources. See :class:`~ray.air.config.ScalingConfig` for more information.
```
Comment (justinvyu) [inline doc/source/tune/tutorials/tune-resources.rst:67]:
```suggestion
```

**Review by justinvyu — APPROVED** [review]

File: doc/source/tune/tutorials/tune-resources.rst [hunk 4]
```diff
@@ -18,6 +18,10 @@ of CPUs (cores) on your machine.
     )
     results = tuner.fit()
 
+.. warning::
+
+    ``tune.with_resources`` cannot be used with :ref:`Ray AIR Trainers <air-trainers>`. If you are passing a Trainer to a Tuner, specify the resource requirements in the Trainer instance using :class:`~ray.air.config.ScalingConfig`. The general principles outlined below still apply.
+
 You can override this per trial resources with ``tune.with_resources``. Here you can
```
Comment (justinvyu) [inline doc/source/tune/tutorials/tune-resources.rst:25]: One more suggestion: we don't actually link to `tune.with_resources` anywhere in this guide:

```suggestion
You can override this per trial resources with :ref:`tune-with-resources`. Here you can
```

**Review by krfricke — APPROVED** [review]
Thanks!

**Pull request merged**